frontend/packages/db-structure/src/parser/schemarb/parser.ts

```diff
@@ -399,6 +414,45 @@ function extractConstraintName(node: KeywordHashNode): string | null {
  */
 function extractCheckConstraint(
   argNodes: Node[],
+): Result<CheckConstraint, UnexpectedTokenWarningError> {
+  // Extract string values (table name and constraint detail)
+  const stringValuesResult = extractStringValues(argNodes)
+  if (stringValuesResult.isErr()) {
+    return err(stringValuesResult.error)
+  }
+
+  const stringValues = stringValuesResult.value
+  if (stringValues.length !== 1) {
+    return err(
+      new UnexpectedTokenWarningError(
+        'Check constraint must have one string of its detail',
+      ),
+    )
+  }
+
+  const [detail] = stringValues as [string]
+
+  // Create constraint with detail
+  const constraint = aCheckConstraint({ detail })
+
+  // Extract constraint name from options if present
+  for (const node of argNodes) {
+    if (node instanceof KeywordHashNode) {
+      const name = extractConstraintName(node)
+      if (name) {
+        constraint.name = name
+      }
+    }
+  }
+
+  return ok(constraint)
+}
+
+/**
+ * Extract check constraint details
+ */
+function extractCheckConstraintWithTableName(
```

Comment:
The logic of `extractCheckConstraint` and `extractCheckConstraintWithTableName` is quite similar, but it's difficult to generalize due to the difference in the number of arguments.